### ISSUE TITLE
weaveDNS API for watching names

### ISSUE THREAD

**Issue by inercia** [weaveworks/weave#1591]
weaveDNS could provide an API for _watching_ names, so a client could keep a connection to weaveDNS and be notified when an IP has been added/removed to/from a name. This would be useful for [other systems](https://github.com/inercia/weave-reverse-proxy).

It could be implemented as a regular REST API with some kind of long-polling (we could be inspired by the _etcd_ protocol, but maybe it would be too complicated and it would make things harder on the client side), or by something like Redis, etc

**Comment by rade** [comment]
Do we expect this feature to be available to application containers?

I think that would be fine conceptually from a security POV - anything you can learn via the event stream you can already discover through A and RR queries.

But it does complicate implementation since a) we'd somehow have to make the event stream endpoint accessible to containers, and b) do so w/o also enabling access to the weave control port.

**Comment by inercia** [comment]
> I think that would be fine conceptually from a security POV - anything you can learn via the event stream you can already discover through A and RR queries.

Yes, I think the same rules should apply for DNS than for this HTTP API. In fact, I would suggest to export this API through a separate HTTP mux at a different port (eg, `1153`), so the same rules we added in #1606 should be added for `1153`.

**Comment by inercia** [comment]
I was also thinking that maybe this API could be compatible with _Consul_, so we could listen for queries about [services](https://www.consul.io/docs/agent/http/catalog.html#catalog_service). The problem is that, while _Consul_ knows about ports, _weaveDNS_ doesn't. But this would make things easier in case users want to switch to a _Weave_-less environment...
